Remove deploy step for macOS CI

vespakoen opened the pull request:
Removing deploy step from macOS CI `test` stage.
Will also add the `if: tag IS present` condition if the first build succeeds.

ppd commented:
Suggestion: commit message `Travis: remove deploy step from macOS CI test stage`

vespakoen commented:
Allright! I will squash those commits before merge.

- I also added OpenMP
- Forced it to build with `BUILD_TYPE=Release` to try that out
- Updated the xcode version in the travis build

Let's see what happens, i'll use this branch to try out some things and hopefully land on a nice setup for macOS soon.

ruevs commented on src/drawentity.cpp:
```diff
@@ -315,9 +315,13 @@ void Entity::ComputeInterpolatingSpline(SBezierList *sbl, bool periodic) const {
-                if(i > 0)     bm.A[i][i - 1] = eq.x;
-                /**/          bm.A[i][i]     = eq.y;
-                if(i < (n-1)) bm.A[i][i + 1] = eq.z;
+                if(i > 0) {
```
@vespakoen @phkahler 
These changes (I am not sure if they make sense) do not belong in this commit!

ruevs commented on src/solvespace.h:
```diff
@@ -513,7 +513,7 @@ class SolveSpaceUI {
 
     // The state for undo/redo
-    typedef struct {
+    typedef struct UndoState {
```
Yes, I've been meaning to make a pull request for this annoying warning.
But, again, in my opinion it does not belong in this his commit.

ruevs commented on src/mouse.cpp:
```diff
@@ -624,24 +624,24 @@ void GraphicsWindow::MouseRightUp(double x, double y) {
         if(gs.withEndpoints > 0) {
             menu->AddItem(_("Select Edge Chain"),
-                          [this]() { MenuEdit(Command::SELECT_CHAIN); });
+                []() { MenuEdit(Command::SELECT_CHAIN); });
```
Same here. This whole file...

ruevs reviewed (commented):
Please be more careful what you include in the pull requests.

vespakoen commented on .travis/build-macos.sh:
```diff
@@ -1,9 +1,23 @@
-make -j2
-make test_solvespace
+
+export BUILD_TYPE=Release
```
This was just for checking if the build would work in Release mode, I changed it back to the default (automatic switching between `Debug` and `RelWithDebInfo` based on the presence of `$TRAVIS_TAG`

ruevs replied:
I noticed too late. Sorry.

vespakoen commented:
@ruevs Points taken, sorry, I will split these up better in the future.

There were a lot of warnings, and I got some LLVM (v11) crashes / parse errors, so I thought maybe removing those warnings would get me further.

phkahler commented:
Wondering if we should unconditionally build with OpenMP now. Maybe the option could disable it - that's sometimes helpful in development. We were not using it before because whitequark didn't want users to need the extra DLL on Windows where we ship with no installer (just an exe). Jonathan is OK with requiring that, as am I. So if we're going to enable it for releases, should we just build with it unless set otherwise?

LTO is different because it takes a lot (3 to 4x)  longer to build and offers only 30% faster code (in the one test I measured). So I still think that has to be for release builds only.
